Remove inherited properties

Traksewt opened the pull request:
Currently the child models were showing properties inherited from the
parent. This was cluttering up the uml for large projects.

Traksewt committed:
remove inherited properties

Currently the child models were showing properties inherited from the
parent. This was cluttering up the uml for large projects.

Traksewt committed:
rename method to look for base properties

Traksewt committed:
remove iml

alemhnan commented:
Hi @Traksewt , 

Many thanks for the PR. Just checked and it looks good. Now we have an option to hide all properties ( triggered by ?properties=false as query string). 
Would you consider adding an option also for the inherited properties so we can switch on/off as well?

Traksewt committed:
adding keyed cache.
adding passing url properties to loopback source generation
adding hideInherited property to URL

Traksewt committed:
handle real boolean from loopback config

Traksewt commented:
Hi @alemhnan,
I didn't see those url properties before. I just changed the code to:
- pass in all URL properties to the loopback source generation. This is because we get to use the rich loopback models, which is easier for handling tasks like looking through superclasses hierarchically than string manipulation on the source file directly. The properties merge in the component config loopback setup, so you can setup defaults as well.

- make the source cache keyed by the options. This way you can request different URL parameters but still make use of the cache where appropriate.

- made the diagram generator support hideInherited boolean option. This works as a URL parameter or as a default setup in the component-config.

alemhnan commented:
Again thanks a lot! I'll merge and publish asap.
